packages/libs/eda/src/lib/core/hooks/workspace.ts

```diff
@@ -116,12 +116,70 @@ export function useStudyEntities(filters?: Filter[]) {
                         : filter.type === 'stringSet'
                         ? filter.stringSet
                         : undefined;
-                    // augment variable metadata
-                    return {
-                      ...variable,
-                      vocabulary,
-                      distinctValuesCount: vocabulary?.length ?? 0,
-                    };
+                    // need to strip 'T00:00:00Z' from filter.min/max
+                    const filterRange =
+                      filter.type === 'numberRange' ||
+                      filter.type === 'dateRange'
+                        ? {
+                            min:
+                              filter.type === 'numberRange'
+                                ? filter.min
+                                : filter.min.split('T00:00:00Z')[0],
+                            max:
+                              filter.type === 'numberRange'
+                                ? filter.max
+                                : filter.max.split('T00:00:00Z')[0],
+                          }
+                        : undefined;
+
+                    // augment variable metadata including filter-aware axis range
+                    if (
+                      variable.type === 'number' ||
+                      variable.type === 'integer'
+                    )
+                      return {
+                        ...variable,
+                        vocabulary,
+                        distinctValuesCount: vocabulary?.length ?? 0,
```

Comment:
Nice!
I don't have time for a full review or test right now, but I wonder if we need to bother augmenting `distinctValuesCount` for range filters?  I don't think the `vocabulary` exists on `type: 'number'` variables, and I don't think we want it to be zero (could that "constrain it away" from some variable pickers?).

We don't know how many values are left after filtering. Would it be useful to know that only 7 distinct values are left? I guess so, it would make it available as a barplot x-axis variable. BUT is that really worth the hassle of finding out how many values pass the filter?  I don't think so.  I think it would be save to leave `distinctValuesCount` unchanged.

@dmfalke second opinion?

Reply:
I agree that we should leave that value unchanged. I don't think it will impact the primary use case, and we can't easily find the correct value.